rotkehlchen/globaldb/updates.py

```diff
@@ -35,12 +37,36 @@ def executeall(cursor: sqlite3.Cursor, statements: str) -> None:
         cursor.execute(statement)
 
 
+def _replace_assets_from_db(
+        connection: sqlite3.Connection,
+        sourcedb_path: Path,
+) -> None:
+    cursor = connection.cursor()
+    cursor.executescript(f"""
+    ATTACH DATABASE "{sourcedb_path}" AS other_db;
+    PRAGMA foreign_keys = OFF;
+    DELETE FROM assets;
+    DELETE FROM ethereum_tokens;
+    DELETE FROM underlying_tokens_list;
+    DELETE FROM common_asset_details;
+    INSERT INTO assets SELECT * FROM other_db.assets;
+    INSERT INTO ethereum_tokens SELECT * FROM other_db.ethereum_tokens;
+    INSERT INTO underlying_tokens_list SELECT * FROM other_db.underlying_tokens_list;
+    INSERT INTO common_asset_details SELECT * FROM other_db.common_asset_details;
+    UPDATE settings SET value=(
+        SELECT value FROM other_db.settings WHERE name="{ASSETS_VERSION_KEY}"
+    ) WHERE name="{ASSETS_VERSION_KEY}";
+    PRAGMA foreign_keys = ON;
+    DETACH DATABASE "other_db";
+    """)
+
+
 def _force_remote(cursor: sqlite3.Cursor, local_asset: Asset, full_insert: str) -> None:
     """Force the remote entry into the database by deleting old one and doing the full insert.
-
+σ
     May raise an sqlite3 error if something fails.
     """
-    cursor.execute('PRAGMA foreign_keys=off;')
+    cursor.executescript('PRAGMA foreign_keys=off;')
```

Comment:
Why is executescript used here? It makes a commit before executing the sql statement. As per discused this morning doesn't it need to be a commit after the pragma sentence?

Reply:
exactly because I need this to be commited.

Reply:
But need to confirm if executescript does a commit at the end or not. I thought it does. If it does not, then I will have to also add a `COMMIT` inside there too.

Reply:
As per the docs it is a commit then sql https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.executescript

Reply:
All right so I will add a commit in there too, just to keep it as one liner.